tests/app/storage/test_datastore.py

```diff
@@ -23,7 +23,7 @@ def setUp(self):
         self.ds = Datastore(self.mock_client)
 
     def test_get_by_key(self):
-        model = QuestionnaireState("someuser", "data", 1)
+        model = QuestionnaireState("someuser", "data", 1, None)
```

Comment:
I think it would make sense if `QuestionnaireState` could be initialised without a `submitted_at` i.e. default to `None`.